consensus/hotstuff/randombeacon_signer.go

```diff
@@ -0,0 +1,57 @@
+package hotstuff
+
+import "github.com/onflow/flow-go/crypto"
+
+type RandomBeaconSigner interface {
+	// Verify verifies whether the signature is from the signer specified by the given signer index
+	// it allows concurrent verification of the given signature
+	// return nil if signature is valid
+	// return crypto.InvalidInputsError if the signature is invalid
+	// return other error if there is exception
+	Verify(signerIndex int, share crypto.Signature) error
+
+	// TrustedAdd adds a verified share to the internal signature shares store
+	// The operation is sequential.
+	// It assumes the signature share has been verified and is valid.
+	// return (true, nil) if the signature has been added
+	// return (false, nil) if the signature is a duplication
+	// return (false, error) if there is any exception
+	// TODO: should we let it also return EnoughShares?, because otherwise
```

Comment:
Generally speaking I would return if it has accumulated enough weight in `TrustedAdd` this makes easier to use it.